pkg/acceptance/bettertestspoc/README.md

```diff
@@ -0,0 +1,460 @@
+```go
+func (w *AbcAssert) HasXyz(expected string) *AbcAssert {
+    w.assertions = append(w.assertions, func(t *testing.T, o *sdk.Abc) error {
+        t.Helper()
+        if o.Xyz != expected {
+            return fmt.Errorf("expected xyz: %v; got: %v", expected, o.Xyz)
+        }
+        return nil
+    })
+    return w
+}
+```
+
+- `abc_snowflake_ext.go` - for the easier separation later (when we start generating the common checks for each object). Example would be:
+```go
+func (w *WarehouseAssert) HasStateOneOf(expected ...sdk.WarehouseState) *WarehouseAssert {
+    w.assertions = append(w.assertions, func(t *testing.T, o *sdk.Warehouse) error {
+        t.Helper()
+        if !slices.Contains(expected, o.State) {
+            return fmt.Errorf("expected state one of: %v; got: %v", expected, string(o.State))
+        }
+        return nil
+    })
+    return w
+}
+```
+
+## Adding new models
+For object `abc` create the following files with the described content in the `config` package:
+- `abc_model.go`
+```go
+type AbcModel struct {
+    Xyz config.Variable `json:"xyz,omitempty"`
+    *resourceModelMeta
+}
+```
+two builders with required params only:
+```go
+func NewAbcModel(
+    resourceName string,
+    xyz string,
+) *AbcModel {
+    m := &AbcModel{resourceModelMeta: meta(resourceName, resources.Abc)}
+    m.WithXyz(xyz)
+    return m
+}
+
+func NewDefaultAbcModel(
+    xyz string,
+) *AbcModel {
+    m := &AbcModel{resourceModelMeta: defaultMeta(resources.Abc)}
+    m.WithXyz(xyz)
+    return m
+}
+```
+Two methods for each param (with good value type and with any value type):
+```go
+func (m *AbcModel) WithXyz(xyz string) *AbcModel {
+    m.Xyz = config.StringVariable(xyz)
+    return m
+}
+
+func (m *AbcModel) WithXyzValue(value config.Variable) *AbcModel {
+    m.Xyz = value
+    return m
+}
+```
+
+- `abc_model_ext.go` - for the easier separation later (when we start generating the models for each object). Example would be:
+```go
+func BasicWarehouseModel(
+    name string,
+    comment string,
+) *WarehouseModel {
+    return NewDefaultWarehouseModel(name).WithComment(comment)
+}
+```
+
+## Known limitations/planned improvements
+- Generate all assertions and models.
+- Test all the utilities for assertion/model construction (public interfaces, methods, functions).
+- Verify if all the config types are supported.
+- Consider a better implementation for the model conversion to config (TODO left).
+- Support additional methods for references in models (TODO left).
+- Support depends_on in models (TODO left).
+- Add a convenience function to concatenate multiple models (TODO left).
+- Add function to support using `ConfigFile:` in the acceptance tests.
+- Replace `acceptance/snowflakechecks` with the new proposed Snowflake objects assertions.
+- Support `showOutputValueUnset` and add a second function for each `show_output` attribute.
+- Support `resourceAssertionTypeValueNotSet` for import checks (`panic` left currently).
```

Comment:
Please add support for `describe_output`.

Reply:
Good one! Added.